### ISSUE TITLE
NuGet Packages and Dependencies

### ISSUE THREAD

**Issue by EatonZ** [GoogleCloudPlatform/google-cloud-dotnet#1790]
**Environment**
-Visual Studio 2015
-.NET 4.6.2 Console Application

When installing a Google .NET package, such as `Google.Apis.CloudKMS.v1`, it wants to install a bunch of dependencies that don't really make sense. In the .NET Framework, these libraries and associated classes are already available, thus downloading these packages seems completely unnecessary.

```
System.Net.Http.4.3.3
System.Security.Cryptography.Algorithms.4.3.1
System.Security.Cryptography.Encoding.4.3.0
System.Security.Cryptography.Primitives.4.3.0
System.Security.Cryptography.X509Certificates.4.3.2
```

Through my research and cleaning, everything still works when the cryptography packages are removed, but there are issues with proxying net traffic without the NuGet version of `System.Net.Http`, so I left that in.

The main issue is when I update `Google.Apis.CloudKMS.v1`, I receive the following extremely uninformative error:
![capture](https://user-images.githubusercontent.com/3020941/34280550-5b0aae54-e686-11e7-8aff-d31c3b0e20fb.PNG)

If the missing packages are installed, the update works fine. So if I add them in and then update, all is well, but I would prefer to update without having to go through this every time.

So in summary, I think you should review your NuGet configuration for the .NET Framework, and get rid of these unnecessary packages. Or, if they are absolutely required and I am mistaken, clarification would be appreciated as well.

**Comment by jskeet** [comment]
Just to check, which Google packages are you using? All the ones maintained in this repo are Google.Cloud.*. If you're only using Google.Apis.*, we should move this over to https://github.com/google/google-api-dotnet-client (I'm happy to create the new issue for you).

It does seem odd that we have a requirement for System.Net.Http, but I know that things get complicated for that particular packages for reasons I don't understand. @chrisdunelm Can you remember the details of that?

I suspect that System.Net.Http is the source of all the rest of the dependencies.

**Comment by chrisdunelm** [comment]
Looks like this is a `Google.Apis` issue, so needs transfering to `google-api-dotnet-client`. Will wait for confirmation from @EatonZ before investigating.

**Comment by EatonZ** [comment]
Oops, I guess it goes there. My impression from the readme in that client repository was that it was being discontinued and that this was the successor project.

**Comment by chrisdunelm** [comment]
Your impression is mostly correct. This repo contains the new `Google.Cloud` client libraries; and is where all (well, almost all) current and future development occurs. However, not all client libraries are available here yet; hence the [`Google.Apis` repo](https://github.com/google/google-api-dotnet-client) is still maintained.

The `Google.Apis.CloudKMS.v1` package you referred is a `Google.Apis` client library. I'll close this issue and open one in the other repo now.
